tools/vol2obj/main.c

```diff
@@ -98,6 +125,110 @@ typedef enum img_fmt_t {
+    "If the -l parameter is not given then only this single frame is processed.\n"                        //
+    "Default value 0.\n",                                                                                 //
+    1 },                                                                                                  //
+  { "--last", "-l",                                                                                       // CL_LAST
+    "The next argument gives the frame number of the last frame to process.\n"                            //
+    "Can be used with -f to process a range of frames from first to last, inclusive.\n",                  //
+    1 },                                                                                                  //
+  { "--output-dir", "-o",                                                                                 // CL_OUTPUT_DIR
+    "The next argument gives the path to a directory to write output files into.\n"                       //
+    "Default is the current working directory.\n",                                                        //
+    1 },                                                                                                  //
+  { "--prefix", "-p",                                                                                     // CL_PREFIX
+    "The next argument gives the prefix to use for output filenames.\n"                                   //
+    "Default is output_frame_.\n",                                                                        //
+    1 },                                                                                                  //
+  { "--sequence", "-s", "Required. The next argument gives the path to the sequence_0.vols file.\n", 1 }, // CL_SEQUENCE
+  { "--video", "-v", "Required. The next argument gives the path to the video texture file.\n", 1 }       // CL_VIDEO
+};
+
+/** Used to print all the options in the command line flags struct for the help text. */
+static void _print_cl_flags( void ) {
+  printf( "Options:\n" );
+  for ( int i = 0; i < CL_MAX; i++ ) {
+    if ( _cl_flags[i].long_str ) { printf( "%s", _cl_flags[i].long_str ); }
+    if ( _cl_flags[i].long_str && _cl_flags[i].short_str ) { printf( ", " ); }
+    if ( _cl_flags[i].short_str ) { printf( "%s", _cl_flags[i].short_str ); }
+    if ( _cl_flags[i].long_str || _cl_flags[i].short_str || _cl_flags[i].short_str ) { printf( "\n" ); }
+    if ( _cl_flags[i].help_str ) { printf( "%s\n", _cl_flags[i].help_str ); }
+  }
+}
+
+static bool _check_cl_option( int argv_idx, const char* long_str, const char* short_str ) {
+  if ( long_str && ( 0 == strcasecmp( long_str, my_argv[argv_idx] ) ) ) { return true; }
+  if ( short_str && ( 0 == strcasecmp( short_str, my_argv[argv_idx] ) ) ) { return true; }
+  return false;
+}
+
+/** If command-line options are valid, their index in argv is stored here, otherwise it is 0. */
```

Comment:
this array keeps track of each option flag - if it's in argv then its corresponding element here gives the argv element it appears in, otherwise 0.